test/test_app.rb

```diff
@@ -190,6 +190,51 @@ def initialize(formatter)
                 end
             end
 
+            describe "lock and unlock log dir" do
+                before do
+                    app.log_base_dir = File.join(make_tmpdir, "log", "path")
+                end
+
+                it "locks directory" do
+                    full_path = app.find_and_create_log_dir("tag")
+                    app.lock_log_dir
+
+                    assert app.log_dir_locked?
+                    assert_equal File.join(full_path, ".lock"), app.lock_file.path
+                end
+
+                it "raises LogDirNotInitialized when trying to lock " \
+                   "an unexisting directory" do
+                    assert_raises(Roby::Application::LogDirNotInitialized) do
+                        app.lock_log_dir
+                    end
+                end
+
+                it "unlocks directory when it is reset" do
+                    app.find_and_create_log_dir("tag")
+                    app.lock_log_dir
+                    app.reset_log_dir
+
+                    refute app.log_dir_locked?
+                end
+
+                it "makes sure locks directory using the temp file" do
+                    full_path = app.find_and_create_log_dir("tag")
+                    temp_lock_file = File.join(full_path, ".lock.tmp")
+                    final_lock_file = File.join(full_path, ".lock")
+
+                    refute File.exist?(final_lock_file)
+                    refute app.log_dir_locked?
+
+                    app.lock_log_dir
+
+                    refute File.exist?(temp_lock_file)
+                    assert File.exist?(final_lock_file)
+                    assert app.log_dir_locked?
```

Comment:
Since log_dir_locked? checks none of the properties of a "locked directory", this is very much NOT a good test.